quantum/pointing_device_drivers.c

```diff
@@ -141,6 +145,20 @@ report_mouse_t cirque_pinnacle_get_report(report_mouse_t mouse_report) {
     if (timer_elapsed(mouse_timer) > (CIRQUE_PINNACLE_TOUCH_DEBOUNCE)) {
         mouse_timer = 0;
     }
+
+    if (!touchData.valid) {
+        return mouse_report;
+    }
+
+    // Scale coordinates to arbitrary X, Y resolution
+    cirque_pinnacle_scale_data(&touchData, cirque_pinnacle_get_scale(), cirque_pinnacle_get_scale());
+
+    if (x && y && touchData.xValue && touchData.yValue) {
+        report_x = (int8_t)(touchData.xValue - x);
+        report_y = (int8_t)(touchData.yValue - y);
+    }
+    x              = touchData.xValue;
+    y              = touchData.yValue;
```

Comment:
Can we move this chunk of code back up above the tap detection?
That way, touchData.valid early return can be at the top, and the scaled x & y coordinate can be used by tap detection if a movement threshold were to be added there.

Reply:
we at least need to check the debounce and reset the timer...

Reply:
That will still happen when there is valid data.
Not sure I understand your concern, can you give me a scenario where it would be a problem?

Reply:
I think I assumed the timer is "an object-like resourde", but it's just a timestamp integer that can be set to 0 at any time. So, no further argument here, see 88a4e3e.

Reply:
Thanks